crates/biome_js_analyze/src/lint/correctness/use_jsx_key_in_iterable.rs

```diff
@@ -206,6 +206,19 @@ fn handle_potential_react_component(
     let node = unwrap_parenthesis(node)?;
 
     if is_inside_jsx {
+        if let AnyJsExpression::JsConditionalExpression(node) = node {
+            let results = [node.consequent().ok()?, node.alternate().ok()?]
+                .into_iter()
+                .filter_map(|node| handle_potential_react_component(node, model, is_inside_jsx))
+                .flatten()
+                .collect::<Vec<_>>();
+
+            return if !results.is_empty() {
```

Comment:
We shouldn't create **a new vector**, only to check if it's empty. Can't we just call the handle function on the two nodes and check both results? A KISS solution fits better in this case

Reply:
Yeah tbh I didn't really like the potential unnecessary heap allocation here.